### PR TITLE
[2/6] Venft data models

### PR DESCRIPTION
**Pull request by jmilldotdev**
## What does this PR do and why?

Adds all constants, data models, and various things necessary to interact with the veBanny data.

Related to https://github.com/jbx-protocol/juice-interface/issues/1416

## Screenshots or screen recordings

No UI updates.

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/constants/veNft/veNftProject.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const VENFT_CONTRACT_ADDRESS =
```
Comment (tomquirk) [inline src/constants/veNft/veNftProject.ts:1]: **question** guessing we can eventually get this from an npm package?

File: src/constants/veNft/veNftResolverAbi.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
+export const veNftResolverAbi = [
```
Comment (tomquirk) [inline src/constants/veNft/veNftResolverAbi.ts:1]: **question** guessing we can eventually get this from an npm package (same as above)?

File: src/utils/notifications.ts [hunk 3]
```diff
@@ -23,3 +23,25 @@ export const emitErrorNotification = (
     },
   })
 }
+
+export const emitSuccessNotification = (
```
Comment (tomquirk) [inline src/utils/notifications.ts:27]: **praise** noice noice

File: src/utils/v2/veNft.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { VeNftVariant } from 'models/v2/veNftVariant'
+
+export const getNFTBaseImage = (
+  baseImagesHash: string,
+  variant: VeNftVariant,
+): string => {
+  const padded = variant.id.toString().padStart(2, '0')
+  return `https://${process.env.NEXT_PUBLIC_PINATA_GATEWAY_HOSTNAME}/ipfs/${baseImagesHash}/${padded}.png`
```
Comment (tomquirk) [inline src/utils/v2/veNft.ts:8]: **question**: this endpoint is heavily rate-limited. I wonder what else we might do here... 🤔

**Review by tomquirk — COMMENTED** [review]
@jmilldotdev sweet, mostly LGTM but I did have one question about the `getNFTBaseImage` fetch.

File: src/constants/veNft/veNftProject.ts (hunk 1 above)
Comment (jmilldotdev) [reply src/constants/veNft/veNftProject.ts:1]: Yes. It is an outstanding issue to move the veNft stuff including ABIs into the JBContracts directory. veNft will also be launched on a per-project basis and fetched similarly to how a project's ERC-20 token is retreived.

File: src/utils/v2/veNft.ts (hunk 4 above)
Comment (jmilldotdev) [reply src/utils/v2/veNft.ts:8]: my idea was to make a single large tiled image, similar to a sprite sheet, and have the frontend load and parse it based on known dimensions.

this was left out of MVP as an optimization step, but we can create an issue to track it.

Comment (jmilldotdev) [reply src/utils/v2/veNft.ts:8]: Partially tracked in #1526, launching and retrieving is part of #1416 which is an epic-level issue that i need to scope and will require discussions with JB contract folks.

File: src/constants/veNft/veNftProject.ts (hunk 1 above)
Comment (jmilldotdev) [reply src/constants/veNft/veNftProject.ts:1]: Partially tracked in https://github.com/jbx-protocol/juice-interface/issues/1526, launching and retrieving is part of https://github.com/jbx-protocol/juice-interface/issues/1416 which is an epic-level issue that i need to scope and will require discussions with JB contract folks.

**Pull request merged**